examples/remote/cargo_workspace/cargo/crates.bzl

```diff
@@ -9,6 +9,162 @@ load("@bazel_tools//tools/build_defs/repo:git.bzl", "new_git_repository")  # bui
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")  # buildifier: disable=load
 load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")  # buildifier: disable=load
 
+# EXPERIMENTAL: A mapping of package names to a set of normal dependencies for the Rust targets of that package.
+_DEPENDENCIES = {
+    "remote/cargo_workspace/num_printer": {
+        "clap": "@remote_cargo_workspace__clap__2_33_3//:clap",
+    },
+    "remote/cargo_workspace/printer": {
+        "ferris-says": "@remote_cargo_workspace__ferris_says__0_2_0//:ferris_says",
+    },
+    "remote/cargo_workspace/rng": {
+        "rand": "@remote_cargo_workspace__rand__0_7_3//:rand",
+    },
+}
+
+# EXPERIMENTAL: A mapping of package names to a set of proc_macro dependencies for the Rust targets of that package.
+_PROC_MACRO_DEPENDENCIES = {
+    "remote/cargo_workspace/num_printer": {
+    },
+    "remote/cargo_workspace/printer": {
+    },
+    "remote/cargo_workspace/rng": {
+    },
+}
+
+# EXPERIMENTAL: A mapping of package names to a set of normal dev dependencies for the Rust targets of that package.
+_DEV_DEPENDENCIES = {
+    "remote/cargo_workspace/num_printer": {
+    },
+    "remote/cargo_workspace/printer": {
+    },
+    "remote/cargo_workspace/rng": {
+    },
+}
+
+# EXPERIMENTAL: A mapping of package names to a set of proc_macro dev dependencies for the Rust targets of that package.
+_DEV_PROC_MACRO_DEPENDENCIES = {
+    "remote/cargo_workspace/num_printer": {
+    },
+    "remote/cargo_workspace/printer": {
+    },
+    "remote/cargo_workspace/rng": {
+    },
+}
```

Comment:
I can't unresolve the other comment because I cannot access it after a rebase.

I view the privacy of these values as separate from the bzl file generation issue, and so I don't think that issue is resolved.

Reply:
That seems unnecessary. I feel that these maps should live in `crates.bzl` until there's a good reason to move them out. I don't see this as any different than the giant macro that gets rendered into the same file for defining each dependency in `genmode="Remote"`. Given that the maps are private and there are experimental notes on each component of the new API, I think it's best to keep everything in one spot.

Reply:
I believe I've made my case very clearly several times, and you have not made an effort to address any of the individual points.

Reply:
> Separate the dynamic component of the generated output from the static component. I think may be easier for readers to understand.

> Put a "soft" guarantee in place that we will not generate "non-data" starlark in the future. I think that's a terrible design rabbit hole.

> Make it easier for future developers to maintain this and keep it formatted (because the files will be simple bzl files, not templates)

My response to all of these is that the change is intended to be a small change that adds an experimental API to try and ease the pain for users who frequently add new crates to their projects. I don't think the state it's in screams to other contributors that they should be following a bad pattern.

> Open the door to (potentially a future change) where cargo-raze consists of both a generation component and a bzl library component.

I'm very interested in working on something like this and before jumping into building out new functionality for two categories of outputs from this tool, I'd like to know that the API works. This is the same thing that `rules_python` does for requirements files. `all_requirements`: https://github.com/bazelbuild/rules_python/blob/master/docs/pip.md#pip_import

Reply:
Marking resolved (given the comments and the explicit opt-in)